Scripts/prepare_release.py

```diff
@@ -0,0 +1,363 @@
+
+class CannotReleaseError(Exception):
+    pass
+
+def progress(msg: str="", **kwargs) -> None:
+    print(msg, **{"file": sys.stderr, "flush": True, **kwargs})
+
+def git(cmd: str, *args: str, **kwargs) -> subprocess.CompletedProcess:
+    return subprocess.run(["git", cmd, *args], # pylint: disable=subprocess-run-check
+        **{"capture_output": True, "check": False, "encoding": "utf-8", **kwargs})
+
+A = TypeVar(name="A")
+def with_error(fn: Callable[[], A]) -> A:
+    try:
+        return fn()
+    except Exception as e:
+        progress(f"failed: {e}")
+        if isinstance(e, subprocess.CalledProcessError):
+            progress("Process output: ")
+            progress(e.stdout)
+            progress(e.stderr)
+        raise CannotReleaseError() from e
+
+def assert_one(msg: str, condition: Callable[[], Any]):
+    assert msg.endswith("?")
+    progress(msg + " ", end="")
+    if not SKIP_ALL_CHECKS and not with_error(condition):
+        progress("no, exiting.")
+        raise CannotReleaseError()
+    progress("yes.")
+
+def run_one(msg: str, operation: Callable[[], None]):
+    assert msg.endswith("...")
+    progress(msg + " ", end="")
+    with_error(operation)
+    progress("done.")
+
+def unwrap(opt: Optional[A]) -> A:
+    assert opt
+    return opt
+
+class NewsFragments:
+    IGNORED = {".gitignore", "README.md"}
+    KNOWN_EXTENSIONS = {".feat": "New features",
+                        ".fix": "Bug fixes"}
+
+    def __init__(self, path: str) -> None:
+        self.path = Path(path)
+        self.fragments: Dict[str, List[Path]] = {}
+        self.unrecognized: List[Path] = []
+        self.collect()
+
+    def collect(self) -> None:
+        if self.path.exists():
+            for f in self.path.iterdir():
+                if f.suffix in self.KNOWN_EXTENSIONS:
+                    self.fragments.setdefault(f.suffix, []).append(f)
+                elif f.name not in self.IGNORED:
+                    self.unrecognized.append(f)
+
+    def check(self):
+        if self.unrecognized:
+            names = ', '.join(quote(f.name) for f in sorted(self.unrecognized))
+            raise ValueError(f"Not sure what to do with {names}.")
+
+    def render(self) -> str:
+        rendered = []
+        for ext, title in self.KNOWN_EXTENSIONS.items():
+            if ext not in self.fragments:
+                continue
+            rendered.append(f"## {title}")
+            for pth in self.fragments[ext]:
+                text = pth.read_text(encoding="utf-8")
+                link = f"(https://github.com/dafny-lang/dafny/pull/{pth.name})"
+                entry = indent(f"- {text}\n{link}", "  ").lstrip()
+                rendered.append(entry)
+        return "\n\n".join(rendered)
+
+    def delete(self):
+        for _, paths in self.fragments.items():
+            for pth in paths:
+                pth.unlink()
+
+class Version(NamedTuple):
+    VERNUM_RE = re.compile("^(?P<prefix>[0-9]+[.][0-9]+[.][0-9]+)(?P<suffix>-.+)?$")
```

Comment:
```suggestion
    VERSION_NUMBER_REGEX = re.compile("^(?P<prefix>[0-9]+[.][0-9]+[.][0-9]+)(?P<suffix>-.+)?$")
```

Reply:
Renamed to `VERSION_NUMBER_PATTERN`, but I find it much harder to read (because the long names hide the rest of the code)

Reply:
Thanks for the renaming, I think it's appropriate. How does the the long name hide the rest of the code? I'm not sure to understand this part.

Reply:
Very long function and variable names make code harder to read, because the relevant bits of syntax / control end up further from each other, at least in my experience.